Configuration.mk

```diff
@@ -109,14 +127,69 @@ ifndef ELF2TAB_EXISTS
   endif
 endif
 
+################################################################################
+##
+## Shared build flags for all architectures in libtock-c.
+##
+################################################################################
+
+# elf2tab flags.
+#
+# Provide the name, memory sizes, and required kernel version as arguments to
+# elf2tab so it can include the parameters in the TBF header.
 ELF2TAB_ARGS += -n $(PACKAGE_NAME)
-ELF2TAB_ARGS += --stack $(STACK_SIZE) --app-heap $(APP_HEAP_SIZE) --kernel-heap $(KERNEL_HEAP_SIZE) --kernel-major $(KERNEL_MAJOR_VERSION) --kernel-minor $(KERNEL_MINOR_VERSION)
+ELF2TAB_ARGS += --stack $(STACK_SIZE) --app-heap $(APP_HEAP_SIZE) --kernel-heap $(KERNEL_HEAP_SIZE)
+ELF2TAB_ARGS += --kernel-major $(KERNEL_MAJOR_VERSION) --kernel-minor $(KERNEL_MINOR_VERSION)
 
-# Setup the correct toolchain for each architecture.
-TOOLCHAIN_cortex-m0 := arm-none-eabi
-TOOLCHAIN_cortex-m3 := arm-none-eabi
-TOOLCHAIN_cortex-m4 := arm-none-eabi
-TOOLCHAIN_cortex-m7 := arm-none-eabi
+# Flags for building app Assembly, C, and C++ files used by all architectures.
+# n.b. make convention is that CPPFLAGS are shared for C and C++ sources
+# [CFLAGS is C only, CXXFLAGS is C++ only]
```

Comment:
```suggestion
# n.b. CPPFLAGS are shared for C and C++ sources (it's short for C Preprocessor,
# and C++ uses the C preprocessor). To specify flags for only C or C++, use
# CFLAGS for C only and CXXFLAGS for C++ only [while we're on the trivia lesson,
# CXX is shorthand for C++ where folks on the unix/gnu side of history needed a
# valid letter rather than a symbol (an X is a rotated +). Confusingly, the dos/microsoft
# lineage chose `.cpp` to address this same issue, leading to confusion nowadays
# about the meaning of 'cpp'.
```

It's not just make convention, there's an actual reason behind all this! I just learned this recently, and I thought it was interesting history. My guess is that anyone digging this deep into the Makefiles will too :D.